llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -1757,6 +1766,43 @@ void VPWidenCastRecipe::execute(VPTransformState &State) {
     setFlags(CastOp);
 }
 
+// Detects whether the extension should be folded away into a combined
+// target instruction, and therefore given a cost of 0.
+// Handles patterns similar to the following:
+//   * partial_reduce(ext, phi)
```

Comment:
Is this supported yet? I do have a patch for this (https://github.com/llvm/llvm-project/pull/133922), but is still in review. I guess the comment doesn't do any harm, but it suggests it's something we already support that's all.